app/services/sign_in/session_revoker.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module SignIn
+  class SessionRevoker
+    attr_reader :refresh_token, :anti_csrf_token, :enable_anti_csrf, :session
+
+    def initialize(refresh_token:, anti_csrf_token:, enable_anti_csrf:)
+      @refresh_token = refresh_token
+      @anti_csrf_token = anti_csrf_token
+      @enable_anti_csrf = enable_anti_csrf
+    end
+
+    def perform
+      anti_csrf_check if enable_anti_csrf
+      find_valid_oauth_session
+      detect_token_theft
+      delete_session!
+    end
+
+    private
+
+    def anti_csrf_check
```

Comment:
Since almost all of these private methods are duplicates of the session refresher ones I'm wondering if there's some way to consolidate them? Since they rely heavily on instance variables maybe they could have a shared parent class like `Session::Revoker` and `Session::Refresher` and those validation logic that takes up the bulk of the class could go there?